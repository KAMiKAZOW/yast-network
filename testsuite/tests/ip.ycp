{

include "testsuite.ycp";
include "network/ip.ycp";

DUMP("check_ip4");
TEST(``(check_ip4(nil)), [], nil);
TEST(``(check_ip4("")), [], nil);
TEST(``(check_ip4("A.1.2.3")), [], nil);
TEST(``(check_ip4("1.2.3")), [], nil);
TEST(``(check_ip4("1..2.34")), [], nil);
TEST(``(check_ip4("1. .2.34")), [], nil);
TEST(``(check_ip4("255.255.256.255")), [], nil);
TEST(``(check_ip4("127")), [], nil);
TEST(``(check_ip4("----------")), [], nil);
TEST(``(check_ip4("127.0.0.1")), [], nil);
TEST(``(check_ip4("255.255.255.0")), [], nil);
TEST(``(check_ip4("255.255.255.255")), [], nil);
TEST(``(check_ip4("192.168.0.1")), [], nil);
TEST(``(check_ip4("195.113.31.123")), [], nil);
TEST(``(check_ip4("255.0.0.1")), [], nil);
TEST(``(check_ip4("10.20.1.29")), [], nil);
TEST(``(check_ip4("1.2.3.4")), [], nil);
TEST(``(check_ip4("0.0.0.0")), [], nil);

DUMP("check_ip6");
TEST(``(check_ip6(nil)), [], nil);
TEST(``(check_ip6("")), [], nil);
TEST(``(check_ip6("----------")), [], nil);
TEST(``(check_ip6("1:1:1:01:1:1:1:1")), [], nil);
TEST(``(check_ip6("fec0:1:2:0:200:1cff:feb5:a7ea")), [], nil);
TEST(``(check_ip6("fe80::200:1cff:feb5:a7ea")), [], nil);
TEST(``(check_ip6("fe80::1cb5:a7ea")), [], nil);

TEST(``(check_ip6("1:2:3:4:5:6:7:8")), [], nil);
TEST(``(check_ip6("::3:4:5:6:7:8")), [], nil);
TEST(``(check_ip6("1:2:3:4:5:6::")), [], nil);
TEST(``(check_ip6("1:2::4:5:6:7:8")), [], nil);
TEST(``(check_ip6("1::3:4:5:6:7:8")), [], nil);
TEST(``(check_ip6("1:2:3:4:5:6::8")), [], nil);
TEST(``(check_ip6("1:2:3::8")), [], nil);
TEST(``(check_ip6("::1")), [], nil);
TEST(``(check_ip6("::1:2")), [], nil);
TEST(``(check_ip6("1:2::")), [], nil);
TEST(``(check_ip6("1::")), [], nil);

TEST(``(check_ip6("0::")), [], nil);
TEST(``(check_ip6("0000::")), [], nil);
TEST(``(check_ip6("0:1::")), [], nil);
TEST(``(check_ip6("1:0::")), [], nil);
TEST(``(check_ip6("1:0::0")), [], nil);

TEST(``(check_ip6("fe80::200:1cff:feb5:5433")), [], nil);

TEST(``(check_ip6(":2:3:4:5:6:7:8")), [], nil);
TEST(``(check_ip6("1:2:3:4:5:6:7:")), [], nil);
TEST(``(check_ip6("1::3:4::6:7:8")), [], nil);
TEST(``(check_ip6("1:2:3:4:5:6:7:8:9")), [], nil);
TEST(``(check_ip6("1:2:3:4::5:6:7:8:9")), [], nil);


DUMP("check_netmask4");
TEST(``(check_netmask4("128.0.0.0")), [], nil);
TEST(``(check_netmask4("192.0.0.0")), [], nil);
TEST(``(check_netmask4("224.0.0.0")), [], nil);
TEST(``(check_netmask4("240.0.0.0")), [], nil);
TEST(``(check_netmask4("248.0.0.0")), [], nil);
TEST(``(check_netmask4("252.0.0.0")), [], nil);
TEST(``(check_netmask4("254.0.0.0")), [], nil);
TEST(``(check_netmask4("255.0.0.0")), [], nil);
TEST(``(check_netmask4("255.128.0.0")), [], nil);
TEST(``(check_netmask4("255.192.0.0")), [], nil);
TEST(``(check_netmask4("255.224.0.0")), [], nil);
TEST(``(check_netmask4("255.240.0.0")), [], nil);
TEST(``(check_netmask4("255.248.0.0")), [], nil);
TEST(``(check_netmask4("255.252.0.0")), [], nil);
TEST(``(check_netmask4("255.254.0.0")), [], nil);
TEST(``(check_netmask4("255.255.0.0")), [], nil);
TEST(``(check_netmask4("255.255.128.0")), [], nil);
TEST(``(check_netmask4("255.255.192.0")), [], nil);
TEST(``(check_netmask4("255.255.224.0")), [], nil);
TEST(``(check_netmask4("255.255.240.0")), [], nil);
TEST(``(check_netmask4("255.255.248.0")), [], nil);
TEST(``(check_netmask4("255.255.252.0")), [], nil);
TEST(``(check_netmask4("255.255.254.0")), [], nil);
TEST(``(check_netmask4("255.255.255.0")), [], nil);
TEST(``(check_netmask4("255.255.255.128")), [], nil);
TEST(``(check_netmask4("255.255.255.192")), [], nil);
TEST(``(check_netmask4("255.255.255.224")), [], nil);
TEST(``(check_netmask4("255.255.255.240")), [], nil);
TEST(``(check_netmask4("255.255.255.248")), [], nil);
TEST(``(check_netmask4("255.255.255.252")), [], nil);
TEST(``(check_netmask4("255.255.255.254")), [], nil);
TEST(``(check_netmask4("255.255.255.255")), [], nil);
TEST(``(check_netmask4("1.2.3.4")), [], nil);
TEST(``(check_netmask4("1.A.3.4")), [], nil);
TEST(``(check_netmask4("0.0.0.0")), [], nil);

TEST(``(check_netmask4(nil)), [], nil);
TEST(``(check_netmask4("")), [], nil);
TEST(``(check_netmask4("33")), [], nil);
TEST(``(check_netmask4("0")), [], nil);
TEST(``(check_netmask4("24")), [], nil);
TEST(``(check_netmask4("32")), [], nil);

DUMP("check_netmask6");
TEST(``(check_netmask6(nil)), [], nil);
TEST(``(check_netmask6("")), [], nil);
TEST(``(check_netmask6("345")), [], nil);
TEST(``(check_netmask6("128")), [], nil);

DUMP("ip2integer");
TEST(``(ip2integer("0.0.0.0")), [], nil);
TEST(``(ip2integer("127.0.0.1")), [], nil);
TEST(``(ip2integer("192.168.110.23")), [], nil);
TEST(``(ip2integer("10.20.1.29")), [], nil);

DUMP("ip2string");
TEST(``(ip2string(0)), [], nil);
TEST(``(ip2string(2130706433)), [], nil);
TEST(``(ip2string(3232263703)), [], nil);
TEST(``(ip2string(169083165)), [], nil);

DUMP("ip2hex");
TEST(``(ip2hex("0.0.0.0")), [], nil);
TEST(``(ip2hex("192.168.1.1")), [], nil);
TEST(``(ip2hex("255.255.255.255")), [], nil);

DUMP("compute_network");
TEST(``(compute_network("127.0.0.1","255.0.0.0")), [], nil);
TEST(``(compute_network("192.168.110.23","255.255.255.0")), [], nil);
TEST(``(compute_network("10.20.1.29","255.255.240.0")), [], nil);

DUMP("compute_broadcast");
TEST(``(compute_broadcast("127.0.0.1","255.0.0.0")), [], nil);
TEST(``(compute_broadcast("192.168.110.23","255.255.255.0")), [], nil);
TEST(``(compute_broadcast("10.20.1.29","255.255.240.0")), [], nil);

}
