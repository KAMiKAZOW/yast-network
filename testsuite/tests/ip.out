Dump	check_ip4
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Dump	check_ip6
Return	false
Return	false
Return	false
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	false
Return	false
Return	false
Return	false
Return	false
Dump	check_netmask4
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	true
Return	true
Return	true
Dump	check_netmask6
Return	false
Return	false
Return	false
Return	true
Dump	ip2integer
Return	0
Return	2130706433
Return	3232263703
Return	169083165
Dump	ip2string
Return	0.0.0.0
Return	127.0.0.1
Return	192.168.110.23
Return	10.20.1.29
Dump	ip2hex
Return	0
Return	C0A80101
Return	FFFFFFFF
Dump	compute_network
Return	127.0.0.0
Return	192.168.110.0
Return	10.20.0.0
Dump	compute_broadcast
Return	127.255.255.255
Return	192.168.110.255
Return	10.20.15.255
Dump	netmask2bits
Return	32
Return	28
Return	27
Return	24
Return	20
Return	12
Return	4
Return	0
Dump	bits2netmask
Return	255.255.255.255
Return	255.255.255.224
Return	255.255.255.240
Return	255.255.255.0
Return	255.255.240.0
Return	255.240.0.0
Return	240.0.0.0
Return	0.0.0.0
